options/posix/generic/unistd-stubs.cpp

```diff
@@ -753,11 +642,7 @@ void swab(const void *__restrict, void *__restrict, ssize_t) {
 }
 
 int symlink(const char *target_path, const char *link_path) {
-	if(!mlibc::sys_symlink) {
-		MLIBC_MISSING_SYSDEP();
-		errno = ENOSYS;
-		return -1;
-	}
+	MLIBC_CHECK_OR_ENOSYS(mlibc::sys_symlink);
```

Comment:
It's redundant to repeat `mlibc::sys_` everywhere. It'd also be best if this macro returned the value of the syscall via statement expressions (and returned out of the function otherwise, so that we can eliminate even that bit of repetition).

Reply:
Returning the function is very ugly, especially for functions that use multiple sysdeps (not every sysdeps is used in only a simple wrapper function)

The `sys_` prefix is redundant but at least it makes grepping easier.

Reply:
In general, the less magic that there is within the macro, the better. The previous solution was a few more lines to type but at least it was obvious what's going on.

Reply:
Fair enough